### PR TITLE
Update to V3

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [ '10', '8', '6' ]
+        node: [ '8', '10', '12', '14' ]
```
Comment (mcollina) [inline .github/workflows/ci.yml:9]: we are dropping node 8

File: package.json [hunk 2]
```diff
@@ -28,19 +29,18 @@
...
     "snazzy": "^8.0.0",
     "standard": "^14.0.2",
     "tap": "^12.6.5"
   },
   "dependencies": {
-    "fastify-plugin": "^1.2.1",
-    "secure-compare": "^3.0.1"
+    "fastify-plugin": "^2.0.0",
+    "secure-compare": "^3.0.1",
+    "tsd": "^0.11.0"
```
Comment (mcollina) [inline package.json:41]: this should not be a dependency.

File: plugin.js [hunk 3]
```diff
@@ -86,7 +86,7 @@ function plugin (fastify, options, next) {
 }
 
 module.exports = fp(plugin, {
-  fastify: '^2.0.0',
+  fastify: '<=3',
```
Comment (mcollina) [inline plugin.js:89]: >= 3.0.0

File: plugin.js [hunk 4]
```diff
@@ -86,7 +86,7 @@ function plugin (fastify, options, next) {
 }
 
 module.exports = fp(plugin, {
-  fastify: '^2.0.0',
+  fastify: '<=3.0.0',
```
Comment (mcollina) [inline plugin.js:89]: this should be 3.x

File: package.json [hunk 5]
```diff
@@ -28,19 +29,18 @@
   },
   "homepage": "https://github.com/fastify/fastify-bearer-auth#readme",
   "devDependencies": {
-    "fastify": "^2.0.0",
+    "fastify": "^3.0.0-alpha.1",
```
Comment (mcollina) [inline package.json:32]: we are on rc1 now

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+	"compilerOptions": {
+		"target": "es6",
+		"module": "commonjs",
+		"noEmit": true,
+		"strict": true,
+		"allowSyntheticDefaultImports": true,
+		"esModuleInterop": true
+	},
+}
```
Comment (mcollina) [inline tsconfig.json:10]: Is this still needed?

**Comment by Ethan-Arrowood** [comment]
ill complete and publish later today

**Comment by Ethan-Arrowood** [comment]
Just realizing now that changes I made to package.json didn't get included in the last commit 😣hotfix coming before I release

**Comment by mashpie** [comment]
this is not published right? As I got `Error: fastify-plugin: fastify-bearer-auth - expected '3.x' fastify version, '2.14.0' is installed` after upgrade to 5.0.0 - downgraded and waiting patently

**Comment by Ethan-Arrowood** [comment]
This was published under v5.0.0 and the `@next` tag on npm. Make sure you update your app's version of fastify as well to `v3.0.0-rc.1`

**Comment by mashpie** [comment]
ah ok, so 3.x is definitely required and

```
module.exports = fp(plugin, {
  fastify: '<=3.0.0',
  name: 'fastify-bearer-auth'
})
```

is not an option? (Stumbled upon in reviews above)

**Comment by mcollina** [comment]
You should use the previous version of this module for fastify v2.

**Comment by mashpie** [comment]
Yep, all good. I was just confused by reviews... and curios :) Thanks!

**Pull request merged**